File: c/resistor-color/resistor_color.c

#include <stddef.h>

#include "resistor_color.h"

static resistor_band_t _colors[] = {
    BLACK,
    BROWN,
    RED,
    ORANGE,
    YELLOW,
    GREEN,
    BLUE,
    VIOLET,
    GREY,
    WHITE
};

unsigned int color_code(resistor_band_t resistorBand)
{
    return resistorBand;
}

const resistor_band_t* colors(void)
{
    return _colors;
}

File: nim/resistor-color/resistor_color.nim
import sequtils
  
type
  ResistorColor * = enum
    Black
    Brown
    Red
    Orange
    Yellow
    Green
    Blue
    Violet
    Grey
    White
  
proc colorCode*(resistorColor: ResistorColor): int =
  ord(resistorColor)

proc colors*(): seq[ResistorColor] =
  toSeq(low(ResistorColor)..high(ResistorColor))